ReactNativeClient/lib/shim-init-node.js

```diff
@@ -412,6 +420,49 @@ function shimInit() {
 	shim.pathRelativeToCwd = (path) => {
 		return toRelative(process.cwd(), path);
 	};
+
+	shim.loadSecureItems = (Setting, items) => {
+		if (shim.isElectron() && !shim.isPortable() && keytar) {
+			return keytar.findCredentials(Setting.constants_.appId).then(credentials => {
+				for (let i = 0; i < credentials.length; i++) {
+					const key = credentials[i].account;
+					let value = credentials[i].password;
+					value = Setting.formatValue(key, value);
+					value = Setting.filterValue(key, value);
+
+					Setting.cache_.push({
+						key: key,
+						value: value,
+					});
+				}
+				Setting.dispatchUpdateAll();
+			});
+		} else {
+			for (let i = 0; i < items.length; i++) {
+				const c = items[i];
+
+				c.value = Setting.formatValue(c.key, c.value);
+				c.value = Setting.filterValue(c.key, c.value);
+
+				Setting.cache_.push(c);
+			}
+			Setting.dispatchUpdateAll();
+		}
```

Comment:
loadSecureItems should only do the absolute minimum related to keychain - it should return an array with the relevant credentials and that's it. There's shouldn't be Setting related code, let alone dispatching actions, in here. That also means refactoring the call in Setting.js

Reply:
Yes, that makes sense. I'll refactor the code as you suggested.